src/backend/marsha/core/tests/test_api_video_live_pairing.py

```diff
@@ -493,3 +496,44 @@ def test_api_video_pairing_challenge_delete_expired(self):
             self.client.post("/api/pairing-challenge")
 
             self.assertEqual(0, LivePairing.objects.count())
+
+    @override_settings(
+        CORS_ALLOWED_ORIGINS=["http://example.com"],
+        CORS_ALLOW_METHODS=["POST", "OPTIONS"],
+    )
+    def test_api_video_pairing_challenge_cors(self):
+        """When CORS is enabled on this url, CORS headers are added to the reponse."""
+        with self.modify_settings(
+            MIDDLEWARE={
+                "append": "corsheaders.middleware.CorsMiddleware",
+            }
+        ):
+            resp = self.client.options(
+                "/api/pairing-challenge",
+                HTTP_ORIGIN="http://example.com",
+                content_type="application/json",
+            )
+
+            self.assertEqual(resp[ACCESS_CONTROL_ALLOW_METHODS], "POST, OPTIONS")
+            self.assertEqual(resp[ACCESS_CONTROL_ALLOW_ORIGIN], "http://example.com")
```

Comment:
Shouldn't we test the response status code ?
And maybe test another endpoint ?

Reply:
I added assertion on the `status_code` but I don't see why I should test an other endpoint. Does this test is not covered by changing the `CORS_URLS_REGEX` setting ? If the endpoint is not matching this regex there is no CORS header

Reply:
Indeed, changing the `CORS_URLS_REGEX` setting covers the usecase I had in mind.

For the status code, my thought was to assert thas a 404 is send for an undefined or an unknown referrer.
Something like
```
    @override_settings(
        CORS_ALLOW_METHODS=["POST", "OPTIONS"],
        CORS_ALLOWED_ORIGINS=["http://example.com"],
        CORS_URLS_REGEX=r"^/api/pairing-challenge$",
    )
    def test_api_video_pairing_challenge_cors_not_matching_urls(self):
        """When CORS is enabled but not matching the current referrer, a 404 should be raised."""
        with self.modify_settings(
            MIDDLEWARE={
                "append": "corsheaders.middleware.CorsMiddleware",
            }
        ):
            response = self.client.options(
                "/api/pairing-challenge",
                HTTP_ORIGIN="http://other-domain.com",
                content_type="application/json",
            )

            self.assertEqual(response.status_code, 404)
```

Reply:
It doesn;t work like this. You will have a status code `200` but without CORS headers. Just like my test not matching `CORS_URLS_REGEX`

Reply:
Ok, so with your feature, the situation reported in following screenshot isn't reproducible anymore ?
![image](https://user-images.githubusercontent.com/720491/146330779-85a3b083-c3c0-4a40-8e5c-3dbdc351b31e.png)
Shouldn't we have a use case when we want to return a 404 ?

Reply:
Ok, I see :+1: